### PR TITLE
Handle survey interruptions

### LINKED ISSUE #147: Handle survey and decision tree interruptions
After other skill's conversation, surveys should not assume that a valid answer for a pending question is theirs.

But what to do? Are there situations where we DO want that to happen? Like

bot: what's your name?
user: why are you asking?
bot: because we need to know for this and that
user: nicolas
bot: thanks nicolas, how old are you?

### PR DESCRIPTION
**Pull request by nekron**
Fixes #147

### CODE REVIEW

File: lib/aida/skills/decision_tree.ex [hunk 1]
```diff
@@ -152,12 +152,12 @@ defmodule Aida.Skill.DecisionTree do
   end
 
   defimpl Aida.Skill, for: __MODULE__ do
-    def init(skill, _bot) do
-      skill
-    end
+    def init(skill, _bot), do: skill
+
+    def wake_up(_skill, _bot, _data), do: :ok
```
Comment (matiasgarciaisaia) [inline lib/aida/skills/decision_tree.ex]: May be best not to change this stuff in the same commit of a feature, so the diffs get easier to read

File: test/aida/decision_tree_test.exs [hunk 2]
```diff
@@ -154,5 +154,38 @@ defmodule Aida.DecisionTreeTest do
...
+      session =
+        session
+        |> Session.merge(%{"language" => "en", ".decision_tree/2a516ba3-2e7b-48bf-b4c0-9b8cd55e003f" => %{"question" => "42cc898f-42c3-4d39-84a3-651dbf7dfd5b"}})
+        |> Session.save()
+
+      message = Message.new("menu", bot, session)
+      message = Bot.chat(message)
+      assert message.reply == ["We have barbecue and pasta"]
+
+      assert message |> Message.get_session(".decision_tree/2a516ba3-2e7b-48bf-b4c0-9b8cd55e003f") == nil
```
Comment (matiasgarciaisaia) [inline test/aida/decision_tree_test.exs]: Should be `nil`, or should _not contain_ the key at all?

File: lib/aida/bot.ex [hunk 3]
```diff
@@ -278,6 +278,8 @@ defmodule Aida.Bot do
   end
 
   def send_message(%{session: session} = message) do
+    session |> Session.save()
```
Comment (matiasgarciaisaia) [inline lib/aida/bot.ex]: Wasn't this needed before?

File: test/aida/decision_tree_test.exs (hunk 2 above)
Comment (nekron) [reply test/aida/decision_tree_test.exs]: The protocol responds with nil when the key is not there. Calling `put_session` with nil will remove the key.

File: lib/aida/bot.ex (hunk 3 above)
Comment (nekron) [reply lib/aida/bot.ex]: It was, but was only on the survey side, it was missing on the scheduled messages side.

File: lib/aida/skills/decision_tree.ex (hunk 1 above)
Comment (nekron) [reply lib/aida/skills/decision_tree.ex]: 👌

**Pull request merged**